applications/tari_dan_wallet_cli/src/command/transaction.rs

```diff
@@ -660,14 +660,20 @@ pub enum CliArg {
     I16(i16),
     I8(i8),
     Bool(bool),
+    Blob(Vec<u8>),
     NonFungibleId(NonFungibleId),
     SubstateAddress(SubstateAddress),
 }
 
 impl FromStr for CliArg {
-    type Err = String;
+    type Err = anyhow::Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if let Some(file) = s.strip_prefix('@') {
```

Comment:
why not use a URL syntax instead:

`data:[<media type>][;base64],<data>`

and if it doesn't start with `data` then assume file path

Reply:
Yeah sounds good, I'll use `data:` `file:` urls and that lets us add support for http etc later (though doubt we'll ever really need that)